src/Umbraco.Core/Models/Content.cs

```diff
@@ -269,7 +269,7 @@ internal void SetPublishInfo(string culture, string name, DateTime date)
             if (_publishInfos == null)
                 _publishInfos = new Dictionary<string, (string Name, DateTime Date)>(StringComparer.OrdinalIgnoreCase);
 
-            _publishInfos[culture] = (name, date);
+            _publishInfos[culture.ToLowerInvariant()] = (name, date);
```

Comment:
@zpqrtbnk do we need this? The dictionary is created with `StringComparer.OrdinalIgnoreCase`

Reply:
we do - 'cos `PublishedCultures` returns `_publishInfos?.Keys` and we want them to be normalized to lowercase